spec.html

```diff
@@ -24176,16 +24314,265 @@ <h1>Runtime Semantics: Evaluation</h1>
+        1. Else,
+          1. Let _leftValue_ be a new empty List.
+        1. Let _rightValue_ be ? DecoratorEvaluation of |Decorator|.
+        1. Prepend _rightValue_ to _leftValue_.
+        1. Return _leftValue_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createdecoratoraccessobject" type="abstract operation">
+      <h1>
+        CreateDecoratorAccessObject (
+          _kind_: ~field~, ~method~, ~accessor~, ~getter~, or ~setter~,
+          _name_: a String or Private Name,
+        ): an Object
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>Returns an object with appropriate accessor functions for the value, for use in a decorator context object.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _accessObj_ be OrdinaryObjectCreate(%Object.prototype%).
+        1. If _kind_ is ~field~, ~method~, ~accessor~, or ~getter~, then
+          1. Let _getterClosure_ be a new Abstract Closure with no parameters that captures _name_ and performs the following steps when called:
+            1. Let _o_ be the *this* value.
+            1. If _name_ is a String, return ? Get(_o_, _name_).
+            1. Else, return ? PrivateGet(_name_, _o_).
+          1. Let _getter_ be CreateBuiltinFunction(_getterClosure_, 0, *""*, &laquo; &raquo;).
+          1. Perform ! CreateDataPropertyOrThrow(_accessObj_, *"get"*, _getter_).
+        1. If _kind_ is ~field~, ~accessor~, or ~setter~, then
+          1. Let _setterClosure_ be a new Abstract Closure with parameters (_value_) that captures _name_ and performs the following steps when called:
+            1. Let _o_ be the *this* value.
+            1. If _name_ is a String, perform ? PrivateSet(_name_, _o_, _value_).
+            1. Else, perform ? Set(_o_, _name_, _value_, *true*).
+            1. Return *undefined*.
+          1. Let _setter_ be CreateBuiltinFunction(_setterClosure_, 1, *""*, &laquo; &raquo;).
+          1. Perform ! CreateDataPropertyOrThrow(_accessObj_, *"set"*, _setter_).
+        1. Return _accessObj_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-createaddinitializerfunction" type="abstract operation">
+      <h1>
+        CreateAddInitializerFunction (
+          _initializers_: a List of ECMAScript function objects,
+          _decorationState_: a Record with a field [[Finished]] (a Boolean),
+        ): a function object
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>Creates a function for a decorator's context object which can be used to add an extra initializer to the class.</dd>
+      </dl>
+      <emu-alg>
+        1. Let _addInitializerClosure_ be a new Abstract Closure with parameters (_initializer_) that captures _initializers_ and _decorationState_ and performs the following steps when called:
+          1. If _decorationState_.[[Finished]] is *true*, throw a *TypeError* exception.
+          1. Append _initializer_ to _initializers_.
```

Comment:
Is this the type of change that requires consensus? Technically it would result in different execution behavior, but I think that it would be correct to do so and don't think passing another type of value would be logically coherent

Reply:
Yes, that would be normative, but i doubt it would be controversial.